### PR TITLE
Expose sire.convert functionality

### PR DESCRIPTION
**Pull request by lohedges**
This PR exposes the new functionality from `sire.convert` through the `BioSimSpace.Convert` sub-package. I have also wrapped functionality to create molecules from SMILES strings, and have exposed this on all wrapped Sire objects, i.e. so you can do things like `system.smiles()` to get a list of SMILES strings for all molecules in the system.

The PR also resolves some minor formatting issues and typos in the documentation, fixes issue #8, and adds support for reading remote trajectory files, as discussed [here](https://github.com/michellab/BioSimSpace/pull/417).

Note that this PR currently requires functionality from the [feature_rdkit2](https://github.com/OpenBioSim/sire/tree/feature_rdkit2) branch of Sire, so I will cancel CI until this is ready. (We will also need a new Sire release, so this PR can be built against that version upwards.) Alternatively I have workarounds that will allow it to work with the current Sire devel, which could be removed at a later date. (Although I would need to disable certain functionality on Windows.)

## Checklist

* I confirm that I have merged the latest version of `devel` into this branch before issuing this pull request (e.g. by running `git pull origin devel`): [y]
* I confirm that I have added a test for any new functionality in this pull request: [y]
* I confirm that I have added documentation (e.g. a new tutorial page or detailed guide) for any new functionality in this pull request: [y]
* I confirm that I have permission to release this code under the GPL3 license: [y]

## Suggested reviewers:
@chryswoods

### CODE REVIEW

**Review by chryswoods — DISMISSED** [review]
This all looks good :-)

**Comment by lohedges** [comment]
Thanks. I'll wait until the Sire devel build is done, then merge. The only thing that I had thought of was adding convenience functions to some of the wrapped Sire objects so that you can do `.toRDKit()`, etc. Perhaps I can just bind these automatically at import time by checking what's available in `BioSimSpace.Convert`.

Oh, does you merge include a release bump, since this PR won't work with the original RDKit feature? It _should_ pull in the most recent package, but who knows. Once there is a new release I can update the Sire requirement.

**Comment by chryswoods** [comment]
No, sorry, I didn't do a version bump. I'll remove the old packages from anaconda.org though, as I want to try and keep it so that we only have one latest devel package available. I agree it may lead to some breakage for people using those packages, but that is the price of living on the bleeding edge ;-)

**Pull request merged**
